buildtools/npm/manifest.go

```diff
@@ -1,28 +1,100 @@
 package npm
 
 import (
+	"path/filepath"
+
 	"github.com/fossas/fossa-cli/errors"
 	"github.com/fossas/fossa-cli/files"
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/pkg"
 )
 
 type Manifest struct {
-	Name    string
-	Version string
-
+	Name         string
+	Version      string
 	Dependencies map[string]string
 }
 
-func FromManifest(filename string) (Manifest, error) {
+// FromManifest creates a manifest from the filepath provided
+func FromManifest(pathElems ...string) (Manifest, error) {
 	var manifest Manifest
-	err := files.ReadJSON(&manifest, filename)
+
+	filePath := filepath.Join(pathElems...)
+
+	err := files.ReadJSON(&manifest, filePath)
 	if err != nil {
 		return Manifest{}, err
 	}
+
 	return manifest, nil
 }
 
-func FromNodeModules(dir string) ([]Manifest, error) {
-	return nil, errors.ErrNotImplemented
+// PackageFromManifest generates a package definition for the provided manifest in the supplied directory. Performs revision resolution
+func PackageFromManifest(pathElems ...string) (pkg.Package, error) {
+	filePath := filepath.Join(pathElems...)
+	manifest, err := FromManifest(filePath)
+	if err != nil {
+		return pkg.Package{}, err
+	}
+
+	manifestAsPackage := convertManifestToPkg(manifest)
+
+	// attempt to resolve revisions if node_modules folder exists and package any imports
+	if len(manifestAsPackage.Imports) == 0 {
```

Comment:
Is this check necessary?

Reply:
small optimization. If we expect 0 imports, we don't need to check for the existence of `node_modules`, saving slightly on i/o